src/adagrams.js

```diff
@@ -1,15 +1,145 @@
+//helper functions
+
+// const randomInt = function (max) {
+//   return Math.floor(Math.random() * max);
+// };
+const randomInt = (max) => Math.floor(Math.random() * max);
+
+const removeLetter = function (arr, target) {
+  let i = arr.indexOf(target);
+  if (i > -1) {
+    arr.splice(i, 1);
+  }
+  return arr;
+};
+
+const uppercaseArray = (str) => {
+  let string = str.toUpperCase();
+  return Array.from(string);
+};
+
+//main functions
+
+//Wave 1
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const lettersObject = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+  //create bag of letters
+  const bagOfLetters = [];
+  Object.keys(lettersObject).forEach((letter) => {
+    for (let i = 0; i < lettersObject[letter]; i++) {
+      bagOfLetters.push(letter);
+    }
+  });
+
+  //draw a hand of 10 random letters
+  const hand = [];
+  while (hand.length !== 10) {
+    let letterIndex = randomInt(bagOfLetters.length);
+    hand.push(bagOfLetters[letterIndex]);
+    removeLetter(bagOfLetters, bagOfLetters[letterIndex]);
+  }
+  return hand;
 };
 
+//wave 2
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const inputArray = uppercaseArray(input);
+  for (let letter of inputArray) {
+    if (lettersInHand.includes(letter) === false) {
+      return false;
+    }
+    removeLetter(lettersInHand, letter);
```

Comment:
We should not be removing letters from `lettersInHand`. This function should only check to see if all of the letters in `input` are in `lettersInHand`. If we're removing letters from `lettersInHand` each time we're just doing a check, the user will run out of letters fast.